pkg/models/commandvalue_test.go

```diff
@@ -569,3 +571,146 @@ func TestNewFloat64Value(t *testing.T) {
 		t.Errorf("NewFloat64Value #2: invalid reading Value: %s", cv.ValueToString())
 	}
 }
+
+type mockStructB struct {
+	ID          string    `json:"id" codec:"id,omitempty"`
+	Pushed      int64     `json:"pushed" codec:"pushed,omitempty"`
+	Device      string    `json:"device" codec:"device,omitempty"`
+	Created     int64     `json:"created" codec:"created,omitempty"`
+	Modified    int64     `json:"modified" codec:"modified,omitempty"`
+	Origin      int64     `json:"origin" codec:"origin,omitempty"`
+	isValidated bool      // internal member used for validation check
+}
+
+// Accepts a mockStructB and returns a CBOR encoded byte array
+func encodeMockEvent(e mockStructB) ([]byte, error) {
+	var handle codec.CborHandle
+	var byteBuffer = make([]byte, 0, 64)
+	enc := codec.NewEncoderBytes(&byteBuffer, &handle)
+	err := enc.Encode(e)
+	if err != nil {
+		return []byte{}, err
+	}
+	return byteBuffer, nil
+}
+
+// Accepts a CBOR encoded byte array and returns mockStructB
+func decodeMockEvent(encodedData []byte) (mockStructB, error) {
+	var err error
+	evt := mockStructB{}
+	err = decodeBinaryValue(bytes.NewReader(encodedData), &evt)
+	return evt, err
+}
```

Comment:
You can probably call `decodeBinaryValue` directly instead of having this pass-thru function if you get rid of `mockStructB`

Reply:
Agree, someone can add such a wrapper later if they find it helps testing of 3rd/4th type.